apps/src/templates/sectionProgress/SectionProgressToggle.jsx

```diff
@@ -21,20 +21,45 @@ class SectionProgressToggle extends React.Component {
     setCurrentView: PropTypes.func.isRequired,
   };
 
+  state = {
+    selectedToggle: this.props.currentView,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    // currentView can be set externally, and the component will
+    // still need to update when that happens.
+    if (this.state.selectedToggle !== nextProps.currentView) {
+      this.setState({selectedToggle: nextProps.currentView});
+    }
+  }
+
   onChange = () => {
-    if (this.props.currentView === ViewType.SUMMARY) {
-      this.props.setCurrentView(ViewType.DETAIL);
+    // Display the toggle based on the internal state so that it is
+    // more immediately responsive. Once setting internal state is
+    // complete, then update the redux currentView.
+    // Timeouts forces a render of the local state before dispatching
+    // the action.
+    if (this.state.selectedToggle === ViewType.SUMMARY) {
+      this.setState({selectedToggle: ViewType.DETAIL}, () => {
+        setTimeout(() => {
+          this.props.setCurrentView(ViewType.DETAIL);
+        }, 0);
```

Comment:
The docs at https://reactjs.org/docs/react-component.html#setstate say:

> The second parameter to setState() is an optional callback function that will be executed once setState is completed and the component is re-rendered.

So, you "shouldn't" need the setTimeout calls here. This may turn out not to be true, but since it's in the docs, I'd recommend at least trying it.

Reply:
That's what I expected too. I've tried, and I just tried again. No luck. 😞

Reply:
Thanks for humoring me. Good to know!